app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,51 @@
+<%= render 'shared/menu'%>
+<%= stylesheet_link_tag "recipe", "data-turbo-track": "reload" %>
+<section class="recipe-show">
+<h1><%= @recipe.name %></h1>
+    <div class="food-container">
+        <div class="food-name">
+            <p>Preparation time:<%= @recipe.preparation_time  %>hours</p>
+            <p>Cooking time: <%= @recipe.cooking_time %></p>
+            <p>Description: <%= @recipe.description%></p>  
+        </div>
+<%= form_for @recipe, url: toggle_public_path(id: @recipe.id), method: :post do |f| %>
+  <div class="swith-btn">
+    <p><%= @recipe.public? ? 'Public' : 'Private' %></p>
+    <label class="switch">
+    <div><%= f.submit "" %></div>
+      <%= f.check_box :public, { checked: @recipe.public } %>
+      <span class="slider round"></span>
+    </label>
+  </div>
+  
+<% end %>
+
+    </div>
+    <div class="recipe-btns">
+        <%= link_to 'Generate Shopping List', shoppinglists_path(), class: 'button' %>
```

Comment:
- [ ] Please make sure that the user can generate a shopping list that has the following functionalities:
![image](https://user-images.githubusercontent.com/62290545/230447687-8657c78d-9cad-430f-99df-a53b0702cb63.png)

Reply:
Sough-ted after solving the association.